scrapy/utils/defer.py

```diff
@@ -267,7 +267,14 @@ def deferred_from_coro(o) -> Any:
             # that use asyncio, e.g. "await asyncio.sleep(1)"
             return ensureDeferred(o)
         # wrapping the coroutine into a Future and then into a Deferred, this requires AsyncioSelectorReactor
-        event_loop = get_asyncio_event_loop_policy().get_event_loop()
+        policy = get_asyncio_event_loop_policy()
+        try:
+            # getting event loop fails when run from separate thread like when using scrapy shell
+            event_loop = policy.get_event_loop()
+        except RuntimeError:
+            # create and set new event loop for this thread
+            event_loop = policy.new_event_loop()
+            asyncio.set_event_loop(event_loop)
```

Comment:
I don’t think this accounts for the `ASYNCIO_EVENT_LOOP`.

See the code were we set the event loop initially: https://github.com/scrapy/scrapy/blob/3a34fa839938d7640e56f3ce12ba60aed31a6ecf/scrapy/utils/reactor.py#L67-L80

I don’t see a clear way to address this, so I suggest:
- We document, in both the documentation of the setting (ASYNCIO_EVENT_LOOP) and that of the command (scrapy shell) that they are not compatible with each other.
- We create a task to make them compatible at some point in the future.
- We include a test here that demonstrates how `ASYNCIO_EVENT_LOOP` is not respected by scrapy shell. Hopefully the existing test for ASYNCIO_EVENT_LOOP helps with that. Funny enough, the test cannot run on Windows, as we use a Linux-specific alternative loop in those tests.

Reply:
You are correct, I didn't even think about that.    I suppose the specificity of the event loop is less important in scrapy shell, since each call to `fetch` get's run in it's own thread.    

An alternative solution to this issue could be to explicitly set the event loop immediately after `Shell.fetch` creates the new thread.  Do you think that would be a more preferrable solution?

This issue is not exclusive to windows, and exists on Linux as well so the testing aspect shouldn't be an problem.

Reply:
> Where should the test go for showing that the setting isn't respected?

I was hoping there would be a way to test this, but after looking into it, I am not sure how to do it myself :sweat: Our ASYNCIO_EVENT_LOOP tests are based on logs, but for the same reason it is not easy to use ASYNCIO_EVENT_LOOP in the first place, we do not log a message about whether or not it is being used.

I think we can skip the testing part.